{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "!awk '$3 == \"gene\" && $7 == \"-\" {print $1\"\\t\"$4\"\\t\"$5}' /rna/myk/analysis/class/guided_mission_bio/binfo1-datapack1/gencode.gtf > /rna/myk/analysis/class/guided_mission_bio/binfo1-datapack1/gencode_negstrand.gtf.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3276124</td>\n",
       "      <td>3741721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3435954</td>\n",
       "      <td>3438772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3445779</td>\n",
       "      <td>3448011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3535200</td>\n",
       "      <td>3537508</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3582674</td>\n",
       "      <td>3584730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27575</th>\n",
       "      <td>chrY</td>\n",
       "      <td>90510316</td>\n",
       "      <td>90542914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27576</th>\n",
       "      <td>chrY</td>\n",
       "      <td>90614770</td>\n",
       "      <td>90617133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27577</th>\n",
       "      <td>chrY</td>\n",
       "      <td>90676615</td>\n",
       "      <td>90678894</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27578</th>\n",
       "      <td>chrY</td>\n",
       "      <td>90763696</td>\n",
       "      <td>90766736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27579</th>\n",
       "      <td>chrY</td>\n",
       "      <td>90850138</td>\n",
       "      <td>90850446</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27580 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chr     start       end\n",
       "0      chr1   3276124   3741721\n",
       "1      chr1   3435954   3438772\n",
       "2      chr1   3445779   3448011\n",
       "3      chr1   3535200   3537508\n",
       "4      chr1   3582674   3584730\n",
       "...     ...       ...       ...\n",
       "27575  chrY  90510316  90542914\n",
       "27576  chrY  90614770  90617133\n",
       "27577  chrY  90676615  90678894\n",
       "27578  chrY  90763696  90766736\n",
       "27579  chrY  90850138  90850446\n",
       "\n",
       "[27580 rows x 3 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "refneg = pd.read_csv(\"/rna/myk/analysis/class/guided_mission_bio/binfo1-datapack1/gencode_negstrand.gtf.bed\", sep=\"\\t\", header=None)\n",
    "refneg.columns = [\"chr\", \"start\", \"end\"]\n",
    "refneg = refneg[refneg[\"chr\"] != \"chrM\"]\n",
    "refneg\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "base = ['A', 'T', 'G', 'C']\n",
    "hexamer_whole_list = [''.join(combination) for combination in itertools.product(base, repeat=6)]\n",
    "\n",
    "motif_counts = {'motif': sevenmer_whole_list, 'motif_down' : [], 'motif_notdown' : []}\n",
    "for motif in sevenmer_whole_list:\n",
    "    motif_down_count = sum(1 for tid in df_down['transcript_id'] if gene_data.get(tid, '').find(motif) != -1)\n",
    "    motif_notdown_count = sum(1 for tid in df_up['transcript_id'] if gene_data.get(tid, '').find(motif) != -1)\n",
    "    motif_counts['motif_down'].append(motif_down_count)\n",
    "    motif_counts['motif_notdown'].append(motif_notdown_count)\n",
    "\n",
    "df_motif = pd.DataFrame(motif_counts)\n",
    "\n",
    "df_motif['notmotif_down'] = len(df_down) - df_motif['motif_down']\n",
    "df_motif['notmotif_notdown'] = len(df_up) - df_motif['motif_notdown']\n",
    "\n",
    "df_motif['motif_down'] = df_motif['motif_down'].astype(int)\n",
    "df_motif['motif_notdown'] = df_motif['motif_notdown'].astype(int)\n",
    "df_motif['notmotif_down'] = df_motif['notmotif_down'].astype(int)\n",
    "df_motif['notmotif_notdown'] = df_motif['notmotif_notdown'].astype(int)\n",
    "\n",
    "df_motif"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "train",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
